### ISSUE TITLE
Publish wheels instead of including binary libraries

### ISSUE THREAD

**Issue by wvengen** [yoctopuce/yoctolib_python#11]
This package contains [binary libraries](https://github.com/yoctopuce/yoctolib_python/tree/master/Sources/cdll) for various platforms in the Python package. This makes the package unnecessarily large, and makes it harder to know what the library sources are exactly (you can find it in the repositories, but that is an extra step, requiring some interpretation). The package is also unusable on systems that are not in the precompiled library architectures (though, perhaps, not such a big need in practice).

Python has a different solution than including binaries, namely a [binary distribution format](https://packaging.python.org/en/latest/specifications/binary-distribution-format/#binary-distribution-format) called wheels. The idea is that there is a source distribution that has a build step (and can include the cpp source code for compilation), and that there are wheels including compiled code for various architectures (and Python versions, if needed). See also the [Python packaging guide](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#packaging-your-project).

Alternatively, the Python source package could be just Python, and use a system library when available (while the wheels could include the library and load it, like [snap7](https://github.com/gijzelaerr/python-snap7)). Or define at build-time what you want, e.g. [matplotlib can switch](https://matplotlib.org/stable/install/dependencies.html#use-system-libraries) between its own and system-provided libraries.

In any case, having a source package and architecture-specific binary wheels would be an improvement.

**Comment by srinsoz** [comment]
Thank you for your suggestion and for taking the time to contribute to our project. We appreciate your feedback and understand the benefits of using wheels for distribution.

We do not think that change is critical. Let me explain:

**Size:** The total size of the current binaries included in the package does not exceed 10Mb, which should not be an issue on modern computers.
For very tiny systems, we believe that using pip is not ideal. Instead, we recommend installing the library from the zip file available on our website or
from GitHub, allowing you to include only the required binary and source files. We have an article on this subject: [How to Start in Python with
Yoctopuce Modules](https://www.yoctopuce.com/EN/article/how-to-start-in-python-with-yoctopuce-modules).

**Platform Compatibility:** Our code is designed to work on any platform. The Python code will ensure that the platform is supported and load the
corresponding binary. If the package is used on a platform not supported by any binaries, it will return an error message that our support team can
use to address and implement support for that platform.

**Future Development:** We are working on a pure Python library. We are at the very beginning of the development, and we do not have a
timeframe for its release. For this library, we will consider your suggestion for the package.

Thank you for your understanding and continued support.

**Comment by wvengen** [comment]
Thank you for your thoughtful response.
Knowing where you are in the development process, I understand your decision now. Good to hear you are consider this as a suggestion as the project matures.

p.s. I did feel the need to [do some library pruning](https://github.com/wvengen/flathub/blob/eef7605cc051a0bbbf531b2546340a4ba25637fb/org.artisan_scope.artisan.yml#L15-L25). When releasing bundled software that include dependencies, things like this can add up. First I tried compiling the yoctopuce cpp library, but that also required messing with yoctolib_python's internals to find the correct library name. Perhaps allowing the use of the library as installed in the system (compiled from the cpp sources), could be an easy middle road. In any case, good luck with the software (and hardware), and thanks for what you've made available.
